src/pages/UserTable/DropDownUserTable.tsx

```diff
@@ -44,130 +48,134 @@ const DropDown = (props: Props) => {
   const [canChangeGoverningBodyAdministration, setCanChangeGoverningBodyAdministration] = useState<boolean>(false);
   const [canChangeUserAccess, setCanChangeUserAccess] = useState<boolean>(false);
   const [canAddDegree, setCanAddDegree] = useState<boolean>(false);
 
+  const [chainOfAccessibility, setChainOfAccessibility] = useState<IDropdownItem>();
+  
+  //Takes only those roles, which can access User Table and 
+  //writes them in array in descending order (as in AdminRole enum)
+  const setUserAdminRoles = (allUserRoles: Array<string>) => {
+
+    const userAdminRoles: Array<AdminRole> = new Array<AdminRole>();
+
+    if (allUserRoles?.includes(Roles.Admin)) {
+      userAdminRoles.push(AdminRole.Admin);
+    }
+    if (allUserRoles?.includes(Roles.GoverningBodyHead)) {
+      userAdminRoles.push(AdminRole.GoverningBodyHead);
+    }
+    if (allUserRoles?.includes(Roles.OkrugaHead)) {
+      userAdminRoles.push(AdminRole.RegionHead);
+    }
+    if (allUserRoles?.includes(Roles.OkrugaHeadDeputy)) {
+      userAdminRoles.push(AdminRole.RegionHeadDeputy);
+    }  
+    if (allUserRoles?.includes(Roles.CityHead)) {
+      userAdminRoles.push(AdminRole.CityHead);
+    }
+    if (allUserRoles?.includes(Roles.CityHeadDeputy)) {
+      userAdminRoles.push(AdminRole.CityHeadDeputy);
+    }
+    if (allUserRoles?.includes(Roles.KurinHead)) {
+      userAdminRoles.push(AdminRole.ClubHead);
+    }      
+    if (allUserRoles?.includes(Roles.KurinHeadDeputy)) {
+      userAdminRoles.push(AdminRole.ClubHeadDeputy);
+    }
+
+    return userAdminRoles;
+  };
+
+  //Takes user Plast role, writes them in array in descending order (as in NonAdminRole enum)
+  const setUserNonAdminRoles = (allUserRoles: Array<string>) => {
+    const userNonAdminRoles: Array<NonAdminRole> = new Array<NonAdminRole>();
+
+    if (allUserRoles?.includes(Roles.FormerPlastMember)) {
+      userNonAdminRoles.push(NonAdminRole.FormerPlastMember);
+    }
+    if (allUserRoles?.includes(Roles.PlastMember)) {
+      userNonAdminRoles.push(NonAdminRole.PlastMember);
+    }
+    if (allUserRoles?.includes(Roles.Supporter)) {
+      userNonAdminRoles.push(NonAdminRole.Supporter);
+    }
+    if (allUserRoles?.includes(Roles.RegisteredUser)) {
+      userNonAdminRoles.push(NonAdminRole.Registered);
+    }
+
+    return userNonAdminRoles;
+  };
+
+  const roles = UserApi.getActiveUserRoles();
+
+  useEffect(() => {
+    const buildChain = async () => {
+      const builder: DropdownItemBuilder = new DropdownItemBuilder();
+      setChainOfAccessibility(builder.build());
+    };
+    buildChain();
+  }, []);
+
+  const lookThroughChain = async () => {
+
+    chainOfAccessibility?.handle(currentUser, setUserAdminRoles(roles), selectedUser, 
+      setUserAdminRoles(selectedUserRoles), setUserNonAdminRoles(selectedUserRoles));
+
+    return chainOfAccessibility?.getHandlersResults() ?? null;
```

Comment:
' ?? null' is useless, beacause it will return null already

Reply:
Done